lib/cdo/python_venv.rb

```diff
@@ -0,0 +1,69 @@
+require 'pty'
+
+module PythonVenv
+  def self.install(*args)
+    # --dev: install dev deps
+    # --prod: don't install dev deps
+    # --frozen-lockfile: don't update the lockfile
+    # --check: fail if lockfile is not up-to-date
+    flags = if ENV['CI']
+              '--dev --frozen-lockfile --check'
+            elsif rack_env? :production
+              '--prod --frozen-lockfile'
+            elsif rack_env? :test
+              '--dev --frozen-lockfile --check'
+            else # staging, adhocs, local environments
+              '--dev'
```

Comment:
@code-dot-org/infrastructure lets say we have two PRs that both modify the package deps, and they both merge and are included in the same staging build. `pdm install` will be run during `rake build`, and this will generate a modified `pdm.lock` (because we aren't running with `--frozen-lockfile` in staging). Will this modified `pdm.lock` be committed similar to how schema.rb changes get recommitted? Do I need to do something intentionally to get that?

Reply:
there are two things going on which seem relevant to this question:

* staging updating schema_cache.yml (if needed) on every build: https://github.com/code-dot-org/code-dot-org/blob/f1e4094a2b14d3e76bd4edbd7c2b42495f47a57a/lib/rake/build.rake#L62-L84
* nightly staging content scoop does not appear to be explicitly excluding schema.rb or any other non-content-related files. you can see from that file's [history](https://github.com/code-dot-org/code-dot-org/commits/staging/dashboard/db/schema.rb) that it sometimes gets updated as part of the staging scoop